base/dict.jl

```diff
@@ -8,7 +8,10 @@ const secret_table_token = :__c782dbf1cf4d6a2e5e3865d7e95634f2e09b5902__
 
 haskey(d::Associative, k) = in(k,keys(d))
 
-function in(p::Tuple{Any,Any}, a::Associative)
+function in(p, a::Associative)
+    if !isa(p,Pair)
+        error("Associative collections only contain Pairs; look for e.g. A=>B instead")
```

Comment:
The descriptive error message seems fine to me, though I'd like it to also point people to `keys` and `values` function, since I think if people are in this situation they are most likely looking for one or the other.